.github/workflows/os-compatibility-test.yaml

```diff
@@ -38,7 +38,7 @@ jobs:
 
       - name: Run make
         run: |
-          make binaries
+          make -j"$(nproc)" binaries
```

Comment:
There was [a discussion on #internal-tools](https://gravitational.slack.com/archives/C0351AUPJ8K/p1695772098098429) about using `-j`, though it was in the context of tests.

Based on that thread, I think we should be safe here as binaries don't seem to have conflicting deps.

Reply:
When I tested it, it worked fine. Let's merge and see if there are any problems with it.